sdk/src/prices/fetchPoolPrices.ts

```diff
@@ -0,0 +1,124 @@
+import { AddressUtil } from "@orca-so/common-sdk";
+import { PublicKey } from "@solana/web3.js";
+import {
+  DecimalsMap,
+  defaultConfig,
+  defaultThresholdConfig,
+  PoolMap,
+  PriceMap,
+  TickArrayMap,
+} from ".";
+import { WhirlpoolContext } from "../context";
+import { SwapUtils, PDAUtil, PoolUtil } from "../utils/public";
+import { convertListToMap, filterNullObjects } from "../utils/txn-utils";
+import { calculatePoolPrices } from "./calculatePoolPrices";
+
+/**
+ * fetchPoolPrices asynchronously fetches the prices for the given mints.
+ * The accounts required ot calculate the prices are fetched naively.
+ * If the accounts have already been fetched, directly invoke calculatePoolPrices.
+ * @param ctx
+ * @param mints
+ * @returns
+ */
+export async function fetchPoolPrices(
+  ctx: WhirlpoolContext,
+  mints: PublicKey[],
+  config = defaultConfig,
+  thresholdConfig = defaultThresholdConfig,
+  refresh = true
+): Promise<PriceMap> {
+  const poolMap = await fetchPoolsForMints(ctx, mints, config, refresh);
+  const tickArrayMap = await fetchTickArraysForPools(ctx, poolMap, config, refresh);
+  const decimalsMap = await fetchDecimalsForMints(ctx, mints, refresh);
+
+  return calculatePoolPrices(mints, poolMap, tickArrayMap, decimalsMap, config, thresholdConfig);
+}
+
+export async function fetchDecimalsForMints(
+  ctx: WhirlpoolContext,
+  mints: PublicKey[],
+  refresh = true
+): Promise<DecimalsMap> {
+  const mintInfos = await ctx.fetcher.listMintInfos(mints, refresh);
+  return mintInfos.reduce((acc, mintInfo, index) => {
+    if (!mintInfo) {
+      throw new Error(`Mint account does not exist: ${mints[index].toBase58()}`);
+    }
+
+    acc[mints[index].toBase58()] = mintInfo.decimals;
+    return acc;
+  }, {} as DecimalsMap);
+}
+
+export async function fetchPoolsForMints(
```

Comment:
TODO: I will modify this to be a more generalizable function to fetch pool accounts from a list of mint addresses